File: src/test/java/com/roscopeco/jasm/model/FloatMathTests.java
package com.roscopeco.jasm.model;

public interface FloatMathTests {
    double testF2d(final float in);
    int testF2i(final float in);
    long testF2l(final float in);

    float testFadd(final float d1, final float d2);
    float testFconst0();
    float testFconst1();
    int testFcmpg(final float a, final float b);
    int testFcmpl(final float a, final float b);

    float testFdiv(final float a, final float b);
    float testFmul(final float a, final float b);
    float testFneg(final float a);
    float testFrem(final float a, final float b);
    float testFsub(final float a, final float b);
}

File: src/test/kotlin/com/roscopeco/jasm/model/FloatMathTests.kt
package com.roscopeco.jasm.model

interface FloatMathTests {
    fun testF2d(input: Float): Double
    fun testF2i(input: Float): Int
    fun testF2l(input: Float): Long
    fun testFadd(d1: Float, d2: Float): Float
    fun testFconst0(): Float
    fun testFconst1(): Float
    fun testFcmpg(a: Float, b: Float): Int
    fun testFcmpl(a: Float, b: Float): Int
    fun testFdiv(a: Float, b: Float): Float
    fun testFmul(a: Float, b: Float): Float
    fun testFneg(a: Float): Float
    fun testFrem(a: Float, b: Float): Float
    fun testFsub(a: Float, b: Float): Float
}